From 00095c0c01fbb093ee2927c396995985d3f65d69 Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:20 +0100
Subject: [PATCH 42/52] Revert "Access sema internals and revert the list of
 unused decls."

This reverts commit cbc2dfb7e4de7e7bee920e5b646b95d31c161a33.
---
 interpreter/llvm-project/clang/include/clang/AST/Decl.h    | 2 --
 interpreter/llvm-project/clang/include/clang/AST/DeclCXX.h | 1 -
 interpreter/llvm-project/clang/include/clang/Sema/Sema.h   | 1 -
 3 files changed, 4 deletions(-)

diff --git a/interpreter/llvm-project/clang/include/clang/AST/Decl.h b/interpreter/llvm-project/clang/include/clang/AST/Decl.h
index 609f818008..f26fb5ad5f 100644
--- a/interpreter/llvm-project/clang/include/clang/AST/Decl.h
+++ b/interpreter/llvm-project/clang/include/clang/AST/Decl.h
@@ -50,8 +50,6 @@
 #include <string>
 #include <utility>
 
-namespace cling { class DeclUnloader; }
-
 namespace clang {
 
 class ASTContext;
diff --git a/interpreter/llvm-project/clang/include/clang/AST/DeclCXX.h b/interpreter/llvm-project/clang/include/clang/AST/DeclCXX.h
index 82bb0b0402..9cebaff63b 100644
--- a/interpreter/llvm-project/clang/include/clang/AST/DeclCXX.h
+++ b/interpreter/llvm-project/clang/include/clang/AST/DeclCXX.h
@@ -265,7 +265,6 @@ class CXXRecordDecl : public RecordDecl {
   friend class DeclContext;
   friend class LambdaExpr;
   friend class ODRDiagsEmitter;
-  friend class ::cling::DeclUnloader;
 
   friend void FunctionDecl::setIsPureVirtual(bool);
   friend void TagDecl::startDefinition();
diff --git a/interpreter/llvm-project/clang/include/clang/Sema/Sema.h b/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
index 4dafd61967..6adb8fb796 100644
--- a/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
+++ b/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
@@ -14173,7 +14173,6 @@ protected:
   friend class ASTReader;
   friend class ASTDeclReader;
   friend class ASTWriter;
-  friend class ::cling::DeclUnloader;
 
 public:
   /// Retrieve the keyword associated
-- 
2.43.5

